zebra-network/src/peer/handshake.rs

```diff
@@ -625,13 +661,7 @@ where
 
         // Spawn a new task to drive this handshake.
         tokio::spawn(fut.instrument(connector_span))
-            // This is required to get error types to line up.
-            // Probably there's a nicer way to express this using combinators.
-            .map(|x| match x {
-                Ok(Ok(client)) => Ok(client),
-                Ok(Err(handshake_err)) => Err(handshake_err.into()),
-                Err(join_err) => Err(join_err.into()),
-            })
+            .map(|x: Result<Result<Client, HandshakeError>, JoinError>| Ok(x??))
```

Comment:
We should try this same pattern whenever we do a `match Result<Result<_>>`

Reply:
Is there a clippy lint for this kind of manual result flattening? Because that would be handy.

Maybe we have to wait for `Result::flatten` or `Result::flatten_into` to stabilise: https://github.com/rust-lang/rust/issues/70142

Reply:
I definitely want `Result::flatten*` in some form 🙏